### PR TITLE
Remove build id from api endpoint

### PR DESCRIPTION
**Pull request by vermakhushboo**
## What does this PR do?

Remove `buildId` from API endpoint and fetch it from deployments document.

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)
Tested manually.

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/functions.php [hunk 1]
```diff
@@ -1448,7 +1449,7 @@
     ->label('sdk.response.model', Response::MODEL_NONE)
     ->param('functionId', '', new UID(), 'Function ID.')
     ->param('deploymentId', '', new UID(), 'Deployment ID.')
-    ->param('buildId', '', new UID(), 'Build unique ID.')
+    ->param('buildId', '', new UID(), 'Build unique ID.', true)
```
Comment (Meldiron) [inline app/controllers/api/functions.php:1452]: I would remove it all-together

Comment (Meldiron) [reply app/controllers/api/functions.php:1452]: Discussed, lets add comment about why its here

File: app/controllers/api/functions.php [hunk 2]
```diff
@@ -1433,7 +1433,8 @@
         $response->noContent();
     });
 
-App::post('/v1/functions/:functionId/deployments/:deploymentId/builds/:buildId')
+App::post('/v1/functions/:functionId/deployments/:deploymentId/build')
+    ->alias('/v1/functions/:functionId/deployments/:deploymentId/builds/:buildId')
     ->groups(['api', 'functions'])
     ->desc('Create build')
```
Comment (Meldiron) [inline app/controllers/api/functions.php:1439]:
```suggestion
    ->desc('Rebuild deployment')
```

File: app/controllers/api/functions.php [hunk 3]
```diff
@@ -1469,6 +1470,9 @@
             throw new Exception(Exception::DEPLOYMENT_NOT_FOUND);
         }
 
+        // buildId is added as param just for backward compatibility
+        // It is fetched from the deployments collection and not from the request as the user doesn't have to know the buildId
+        $buildId = $deployment->getAttribute('buildId', '');
         $build = Authorization::skip(fn () => $dbForProject->getDocument('builds', $buildId));
 
         if ($build->isEmpty()) {
```
Comment (christyjacob4) [inline app/controllers/api/functions.php:1478]: Why do we need this block of code? If you look at the code below, we don't use this build's ID for anything

File: app/controllers/api/functions.php (hunk 2 above)
Comment (vermakhushboo) [reply app/controllers/api/functions.php:1439]: @Meldiron what should be the sdk method called?

File: app/controllers/api/functions.php (hunk 3 above)
Comment (vermakhushboo) [reply app/controllers/api/functions.php]: Removed the code as it wasn't being used anywhere

**Review by Meldiron — APPROVED** [review]

File: app/controllers/api/functions.php [hunk 4]
```diff
@@ -1448,15 +1449,14 @@
     ->label('sdk.response.model', Response::MODEL_NONE)
     ->param('functionId', '', new UID(), 'Function ID.')
     ->param('deploymentId', '', new UID(), 'Deployment ID.')
-    ->param('buildId', '', new UID(), 'Build unique ID.')
+    ->param('buildId', '', new UID(), 'Build unique ID.', true) // added as optional param for backward compatibility
```
Comment (stnguyen90) [inline app/controllers/api/functions.php:1452]: What backwards compatibility are we worried about?

This is scoped for 1.6 which means we're going to be bumping the SDKs major version so breaking changes for that should be fine.

Comment (christyjacob4) [reply app/controllers/api/functions.php:1452]: The endpoint should continue to work for people using older SDKs as well

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**